### PR TITLE
Add twitter clone demo

### PR DESCRIPTION
**Pull request by jettify**
screen shot 2017-10-28 at 00 43 46

### CODE REVIEW

File: demos/motortwit/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+clean:
+	rm -rf `find . -name __pycache__`
+	rm -f `find . -type f -name '*.py[co]' `
+	rm -f `find . -type f -name '*~' `
+	rm -f `find . -type f -name '.*~' `
+	rm -f `find . -type f -name '@*' `
+	rm -f `find . -type f -name '#*#' `
+	rm -f `find . -type f -name '*.orig' `
+	rm -f `find . -type f -name '*.rej' `
```
Comment (kxepal) [inline demos/motortwit/Makefile]: find has a `-delete` argument

File: demos/motortwit/motortwit/db.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+follower = t.Dict({
+    t.Key('_id'): MongoId(),
+    t.Key('who_id'): MongoId(),
+    t.Key('whom_id'): t.List(MongoId()),
+})
+
+
+async def get_user_id(user_collection, username):
+    rv = await (user_collection
+                .find_one({'username': username}, {'_id': 1}))
```
Comment (kxepal) [inline demos/motortwit/motortwit/db.py]: quite strange formatting. why not:
```
rv = await user_collection.find_one(
  {'username': username},
  {'_id': 1}
)
```

Comment (kxepal) [reply demos/motortwit/motortwit/db.py]: Also, what is `1` for `_id`?

File: demos/motortwit/motortwit/generate_data.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+# TODO: validate using trafarets before insert
+async def insert_data(collection, values):
+    await collection.insert(values)
+    return values
+
+
+async def generate_users(mongo, schema, rows, fake):
+    values = []
+    pw_hash = generate_password_hash('123456')
```
Comment (kxepal) [inline demos/motortwit/motortwit/generate_data.py]: May be this magic number should come from config?

File: demos/motortwit/motortwit/utils.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import pytz
+import os
+import yaml
```
Comment (kxepal) [inline demos/motortwit/motortwit/utils.py]: I think you should run isort over codebase. Imports order is quite vary over the project.

File: demos/motortwit/static/js/config.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                msg += '</code>';
+            }
+            return msg;
+        });
+
+        var user = nga.entity('user').identifier(nga.field('_id'));
+        var message = nga.entity('message').identifier(nga.field('_id'));
+        var follower = nga.entity('follower').identifier(nga.field('_id'));
+
+        admin
```
Comment (kxepal) [inline demos/motortwit/static/js/config.js]: Shouldn't it be
```
admin.addEntity(user)
     .addEntity(message)
     .addEntity(follower);
```
?

File: demos/motortwit/static/js/config2.js [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            nga.field('_id', 'string'),
+            nga.field('whom_id', 'json'),
+            
+        ]);
+
+            
+    follower.deletionView()
+        .title('Deletion confirmation for entity follower');
+
+
```
Comment (kxepal) [inline demos/motortwit/static/js/config2.js]: too much empty space

File: demos/motortwit/static/js/config2.js [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+            
+    follower.deletionView()
+        .title('Deletion confirmation for entity follower');
+
+        
+
+        nga.configure(admin);
+    }]);
+
+}());
```
Comment (kxepal) [inline demos/motortwit/static/js/config2.js]: new line missed

File: demos/motortwit/motortwit/main.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+def setup_jinja(app):
+    jinja_env = aiohttp_jinja2.setup(
+        app, loader=jinja2.FileSystemLoader(str(TEMPLATES_ROOT)))
+
+    jinja_env.filters['datetimeformat'] = format_datetime
+    jinja_env.filters['robo_avatar_url'] = robo_avatar_url
+
+
+async def init(loop):
+    conf = load_config(str(PROJ_ROOT / 'config' / 'config.yml'))
```
Comment (kxepal) [inline demos/motortwit/motortwit/main.py]: no need to cast Path back to string, `open` can handle it.

File: demos/motortwit/motortwit/db.py (hunk 2 above)
Comment (jettify) [reply demos/motortwit/motortwit/db.py]: fetch only `_id` from document, same as `SELECT user_id FROM users WHERE ...`

File: demos/motortwit/motortwit/generate_data.py (hunk 3 above)
Comment (jettify) [reply demos/motortwit/motortwit/generate_data.py]: this is dummy password for each generated user, not read need to store in config.

File: demos/motortwit/static/js/config.js (hunk 5 above)
Comment (jettify) [reply demos/motortwit/static/js/config.js]: sorry, those js files are not used, removing

**Comment by jettify** [comment]
@kxepal Thanks for review! I addressed your comments, feel free to raise issue I will fix it.

**Pull request merged**